tests/execution_engine/test_pandas_execution_engine.py

```diff
@@ -112,21 +135,52 @@ def test_get_domain_records_with_multicolumn_domain():
             "c": [1, 2, 3, 4, None, 6],
         }
     )
+    # Loading batch data
+    engine.load_batch_data(batch_data=df, batch_id="1234")
+
+    data = engine.get_domain_records(
+        domain_kwargs={
+            "column_list": ["a", "c"],
+            "row_condition": "b>2",
+            "condition_parser": "pandas",
+            "ignore_row_if": "all_values_are_missing",
+        }
+    )
+    data = data.astype(int)
+
     expected_multicolumn_df = pd.DataFrame(
         {"a": [2, 3, 4, 5], "b": [3, 4, 5, 7], "c": [2, 3, 4, 6]}, index=[0, 1, 2, 4]
     )
 
+    assert data.equals(
+        expected_multicolumn_df
+    ), "Data does not match after getting full access compute domain"
+
+    engine = PandasExecutionEngine()
+    df = pd.DataFrame(
+        {
+            "a": [1, 2, 3, 4, 5, 6],
+            "b": [2, 3, 4, 5, None, 6],
+            "c": [1, 2, 3, 4, 5, None],
+        }
+    )
     # Loading batch data
     engine.load_batch_data(batch_data=df, batch_id="1234")
+
     data = engine.get_domain_records(
         domain_kwargs={
-            "column_list": ["a", "c"],
-            "row_condition": "b>2",
+            "column_list": ["b", "c"],
+            "row_condition": "a<6",
             "condition_parser": "pandas",
-            "ignore_row_if": "all_values_are_missing",
+            "ignore_row_if": "any_value_is_missing",
```

Comment:
Love this. It's certainly a more-effective test

Reply:
@cdkini Correct -- the Spark focused PR is coming next.